Working version of workflow (`v0.0.3`)

vinisalazar opened the pull request:
If merged, this PR will bring the latest version of the workflow. There are a few things to fix but overall it's in pretty good shape.

**Summary of changes:**
- Add rule 'blast'
- Add filter_blast_6_containments
- Use blast as ground truth and mmseqs2 as "alternative" for performance generation
- Run black on scripts
- Run workflow formatter and linter
- Improve helper functions
- Collect log on synteny rule

vinisalazar committed:
Add filter command to mmseqs2

  - Apply snakefmt
  - Add filter script

vinisalazar committed:
Add rule 'makeblastdb'

  - Creates blastdb from reference contigs

vinisalazar committed:
Collect log of RScript

vinisalazar committed:
Debug rule 'makeblastdb'

vinisalazar committed:
Add rule 'blastdb'

  - Debug other rules
  - Fix samtools threads problem
  - Use blastn output as ground truth
  - Use mmseqs2 output as submission

vinisalazar committed:
Add gzipped compression to read_csv calls in report script

vinisalazar committed:
Apply linter and snakefmt

vinisalazar committed:
Debugging rule samtools_faidx

vinisalazar committed:
Add RScript dependency

vinisalazar committed:
Fix Rscript call

vinisalazar committed:
Fix ARGS object in synteny script

vinisalazar committed:
Fix get_fai common function

vinisalazar committed:
Add more log calls

vinisalazar committed:
Add multiple makeblastdb outputs

  - Apply snakefmt

vinisalazar committed:
Add pandas to dependencies

vinisalazar committed:
Fix compression format

vinisalazar committed:
Add pattern to list.files call in R script

vinisalazar committed:
Fix blast 6 filter outfmt

vinisalazar committed:
Fix arg in synteny script

vinisalazar committed:
Use correct outputs for performance report script

  - Run black on script

vinisalazar committed:
Run black on filter_blast script

vinisalazar committed:
Fix bool header error

vinisalazar committed:
Update README.md

  - Add link to Manual.md

vinisalazar committed:
Merge remote-tracking branch 'origin/benchmarking' into benchmarking-dev

pghosh2 reviewed (changes requested):
The changes look good to me. Just wanted to point out that there might a merge conflict with some of Andrew's changes in PR#12 with files common.smk, main.smk, performance_report.py.

vinisalazar commented:
Thanks for the review @pghosh2. @ajtritt maybe we could merge #12 to benchmarking and then fix the conflicts on this one ?

vinisalazar commented:
I would suggest changing the base branch of #12 from `main` to `benchmarking`. I did that earlier just to see how it would look like but switched back right after.

vinisalazar committed:
Debugging filtering step

  - Use --as_decimal flag for mmseqs2 filtering
  - Use ctgs.get(s, 0) to avoid KeyErrors in performance_report.py

vinisalazar committed:
Debug synteny script

vinisalazar committed:
Merge branch 'benchmarking' into benchmarking-dev

ajtritt commented on benchmarking/workflow/scripts/performance_report.py:
```diff
@@ -61,16 +61,15 @@ def metrics_by_qual(true, pred):
-    true_df = pd.read_csv(args.true_tsv, sep='\t')
-    pred_df = pd.read_csv(args.pred_tsv, sep='\t')
-
+    true_df = pd.read_csv(args.true_tsv, sep="\t", compression="gzip")
+    pred_df = pd.read_csv(args.pred_tsv, sep="\t", compression="gzip")
```
```suggestion
    true_df = pd.read_csv(args.true_tsv, sep="\t")
    pred_df = pd.read_csv(args.pred_tsv, sep="\t")
```

ajtritt committed:
Update benchmarking/workflow/scripts/performance_report.py
